components/dashboard/src/components/AuthorizeGit.tsx

```diff
@@ -0,0 +1,99 @@
+import { FC, useContext } from "react";
+import { Link } from "react-router-dom";
+import { useAuthProviders } from "../data/auth-providers/auth-provider-query";
+import { useCurrentOrg } from "../data/organizations/orgs-query";
+import { openAuthorizeWindow } from "../provider-utils";
+import { getGitpodService } from "../service/service";
+import { UserContext, useCurrentUser } from "../user-context";
+import { Button } from "./Button";
+import { Heading2, Subheading } from "./typography/headings";
+
+export function useNeedsGitAuthorization() {
+    const authProviders = useAuthProviders();
+    const user = useCurrentUser();
+    if (!user || !authProviders.data) {
+        return false;
+    }
+    return !authProviders.data.some((ap) => user.identities.some((i) => ap.authProviderId === i.authProviderId));
+}
+
+export function AuthorizeGit() {
+    const { setUser } = useContext(UserContext);
+    const org = useCurrentOrg();
+    const authProviders = useAuthProviders();
+    const updateUser = () => {
+        getGitpodService().server.getLoggedInUser().then(setUser);
+    };
+
+    const connect = (ap: AuthProviderInfo) => {
+        openAuthorizeWindow({
+            host: ap.host,
+            scopes: ap.requirements?.default,
+            overrideScopes: true,
+            onSuccess: updateUser,
+        });
+    };
+
+    if (authProviders.data === undefined) {
+        return <></>;
+    }
+
+    const verifiedProviders = authProviders.data.filter((ap) => ap.verified);
+
+    return (
+        <div className="w-full text-center bg-gray-100 dark:bg-gray-800 p-4 m-4 rounded-lg py-10">
+            <Heading2 className="pb-6">Authorize Git</Heading2>
+            {verifiedProviders.length === 0 ? (
+                <>
+                    <Text>There are no Git providers configured.</Text>
+                    {!!org.data?.isOwner ? (
+                        <Text>
+                            <Link className="gp-link" to="/settings/git">
+                                Configure a Git provider
+                            </Link>{" "}
+                            for this organization.
+                        </Text>
+                    ) : (
+                        <Text>Please contact your administrator to configure a Git provider.</Text>
+                    )}
+                </>
+            ) : (
+                <>
+                    <Text>Authorize with {enumerate(verifiedProviders.map((a) => a.host))} to open repositories.</Text>
+                    {verifiedProviders.map((ap) => {
+                        return (
+                            <Button className="mt-5" onClick={() => connect(ap)}>
+                                Authorize with {ap.host}
+                            </Button>
+                        );
+                    })}
+                </>
+            )}
+        </div>
+    );
+}
+
+const Text: FC<{ className?: string }> = ({ className, children }) => {
```

Comment:
Seems like we could use `<Subheading>` directly instead of this `<Text>` wrapper component? If we want it styled slightly differently and easy to use for any text we have, we could add a new Text component to the `components/typography` collection. Probably wouldn't make sense to add unless it looks different than `<Subheading/>` though.